app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java

```diff
@@ -0,0 +1,102 @@
+ */
+
+package com.google.droidjump;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static com.google.droidjump.GameConstants.FIRST_LEVEL_ID;
+import static org.hamcrest.Matchers.not;
+
+import android.content.Intent;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.rule.ActivityTestRule;
+import com.google.droidjump.models.LevelManager;
+import com.google.droidjump.models.NavigationHelper;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class GameSuccessScreenTest {
+    private static final Intent MAIN_ACTIVITY_INTENT =
+            new Intent(InstrumentationRegistry.getInstrumentation().getTargetContext(), MainActivity.class);
+    @Rule
+    public ActivityTestRule<MainActivity> activityTestRule = new ActivityTestRule<>(MainActivity.class);
+
+    @Before
+    public void setUp() throws Exception {
+        activityTestRule.launchActivity(MAIN_ACTIVITY_INTENT);
+        LevelManager.init(activityTestRule.getActivity());
+    }
+
+    @Test
+    public void checkIfSuccessTextIsDisplayed() {
+        NavigationHelper.navigateToFragment(activityTestRule.getActivity(), new GameSuccessFragment());
+        onView(withId(R.id.success_text)).check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void checkOnClickMenuButton() {
+        NavigationHelper.navigateToFragment(activityTestRule.getActivity(), new GameSuccessFragment());
+
+        onView(withId(R.id.menu_button)).check(matches(isDisplayed()));
+        onView(withId(R.id.menu_button)).perform(click());
+
+        onView(withId(R.id.play_button)).check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void playNextLevel() {
+        LevelManager.setCurrentLevelIndex(FIRST_LEVEL_ID);
+        LevelManager.setLastLevelIndex(FIRST_LEVEL_ID);
+        NavigationHelper.navigateToFragment(activityTestRule.getActivity(), new GameSuccessFragment());
+
+        onView(withId(R.id.next_button)).check(matches(isDisplayed()));
+        onView(withId(R.id.next_button))
+                .perform(click());
+
+        onView(withId(R.id.game_layout)).check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void checkNextButtonIsDisplayedIfNextLevelIsAvailable() {
+        LevelManager.setCurrentLevelIndex(FIRST_LEVEL_ID);
+        LevelManager.setLastLevelIndex(FIRST_LEVEL_ID);
+        NavigationHelper.navigateToFragment(activityTestRule.getActivity(), new GameSuccessFragment());
+
+        if (LevelManager.getLevelsLastIndex() > 0) {
```

Comment:
why do we have this if check?
can we use something like asserThat(LevelManager.getLevelsLastIndex()>0).isTrue or something like that?

Reply:
The user can click on "Next button" and play next level in case if there are any levels (levelsCount > 1). levelsCount = LevelManager.getLevelsLastIndex() + 1. if (LevelManager.getLevelsLastIndex() > 0) we need to show "Next Button", in other cases we do not show the button.

Reply:
Then let's have a test for each case. It is better to avoid these things in tests as bugs may be not detected

Reply:
Fixed.